sys/kern/subr_prf.c

```diff
@@ -176,15 +176,24 @@ uprintf(const char *fmt, ...)
 	int retval;
 
 	td = curthread;
-	if (TD_IS_IDLETHREAD(td))
+	if (TD_IS_IDLETHREAD(td)) {
+		/* XXXAR: for debugging early boot failures */
+		va_start(ap, fmt);
+		retval = vprintf(fmt, ap);
+		va_end(ap);
 		return (0);
+	}
 
 	sx_slock(&proctree_lock);
 	p = td->td_proc;
 	PROC_LOCK(p);
 	if ((p->p_flag & P_CONTROLT) == 0) {
 		PROC_UNLOCK(p);
 		sx_sunlock(&proctree_lock);
+		/* XXXAR: for debugging early boot failures */
```

Comment:
If you wanted an upstreamable version, I would perhaps make this vprintf conditional on `if (p == initproc)` and drop the one for idle threads (idle threads should never trigger `uprintf` anyway).   Also, it looks a bit odd to return `0` after getting `retval` from `vprintf`.  I would either ignore `vprintf`'s return value or return `retval`.